kasa/experimental/smartcameraprotocol.py

```diff
@@ -63,37 +75,63 @@ async def close(self) -> None:
         """Close the underlying transport."""
         await self._transport.close()
 
+    @staticmethod
+    def _get_smart_camera_single_request(
+        request: dict[str, dict[str, Any]],
+    ) -> tuple[str, str, str, dict]:
+        method = next(iter(request))
+        if method == "multipleRequest":
+            method_type = "multi"
+            params = request["multipleRequest"]
+            req = {"method": "multipleRequest", "params": params}
+            return "multi", "multipleRequest", "", req
+
+        if (short_method := method[:3]) and short_method in {"get", "set"}:
+            method_type = short_method
+            param = next(iter(request[method]))
+            if method in GET_METHODS_AS_DO:
+                method_type = "do"
+            req = {
+                "method": method_type,
+                param: request[method][param],
+            }
+        else:
+            method_type = "do"
+            param = next(iter(request[method]))
+            req = {
+                "method": method_type,
+                param: request[method][param],
+            }
+        return method_type, method, param, req
+
     async def _execute_query(
         self, request: str | dict, *, retry_count: int, iterate_list_pages: bool = True
     ) -> dict:
         debug_enabled = _LOGGER.isEnabledFor(logging.DEBUG)
-
         if isinstance(request, dict):
             if len(request) == 1:
-                method = next(iter(request))
-                if method == "multipleRequest":
-                    params = request["multipleRequest"]
-                    req = {"method": "multipleRequest", "params": params}
-                elif method[:3] == "set":
-                    params = next(iter(request[method]))
-                    req = {
-                        "method": method[:3],
-                        params: request[method][params],
-                    }
-                else:
-                    return await self._execute_multiple_query(request, retry_count)
+                method_type, method, param, single_request = (
+                    self._get_smart_camera_single_request(request)
+                )
             else:
                 return await self._execute_multiple_query(request, retry_count)
         else:
             # If method like getSomeThing then module will be some_thing
             method = request
+            method_type = request[:3]
             snake_name = "".join(
-                ["_" + i.lower() if i.isupper() else i for i in method]
+                ["_" + i.lower() if i.isupper() else i for i in request]
             ).lstrip("_")
```

Comment:
I'd extract the snake-case conversion into its own helper function and just call it here and elsewhere where it's needed.

Reply:
Done